### PR TITLE
port onchanges to API 8.0 to fix #22

### LINKED ISSUE #22: framework_agreement_sourcing - errors in onchanges
Using onchange raise an AttributesError

```
Traceback (most recent call last):
  File ".../parts/server/openerp/http.py", line 500, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File ".../parts/server/openerp/http.py", line 517, in dispatch
    result = self._call_function(**self.params)
  File ".../parts/server/openerp/http.py", line 283, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File ".../parts/server/openerp/service/model.py", line 113, in wrapper
    return f(dbname, *args, **kwargs)
  File ".../parts/server/openerp/http.py", line 280, in checked_call
    return self.endpoint(*a, **kw)
  File ".../parts/server/openerp/http.py", line 733, in __call__
    return self.method(*args, **kw)
  File ".../parts/server/openerp/http.py", line 376, in response_wrap
    response = f(*args, **kw)
  File ".../parts/server/addons/web/controllers/main.py", line 948, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File ".../parts/server/addons/web/controllers/main.py", line 940, in _call_kw
    return getattr(request.registry.get(model), method)(request.cr, request.uid, *args, **kwargs)
  File ".../parts/server/openerp/api.py", line 237, in wrapper
    return old_api(self, *args, **kwargs)
  File ".../parts/server/openerp/api.py", line 360, in old_api
    result = method(recs, *args, **kwargs)
  File ".../parts/server/openerp/models.py", line 5711, in onchange
    record._onchange_eval(name, field_onchange[name], result)
  File ".../parts/server/openerp/models.py", line 5645, in _onchange_eval
    method_res = getattr(self._model, method)(*args)
  File ".../parts/purchase-workflow/framework_agreement/utils.py", line 30, in wrapper
    return fun(*args, **kwargs)
  File ".../parts/vertical-ngo/framework_agreement_sourcing/model/logistic_requisition_source.py", line 326, in onchange_quantity
    return self.onchange_quantity_obs(cr, uid, ids, qty, date,
AttributeError: 'logistic.requisition.source' object has no attribute 'onchange_quantity_obs'
```

### PR DESCRIPTION
**Pull request by yvaucher**

### CODE REVIEW

**Comment by yvaucher** [comment]
I have to refactor this a bit -> WIP

File: framework_agreement_sourcing/model/logistic_requisition_source.py [hunk 1]
```diff
@@ -289,50 +288,64 @@ def onchange_sourcing_method(self, cr, uid, ids, method, req_line_id,
...
+            return
+        price = agreement.get_price(self.proposed_qty,
+                                    currency=self.currency_id)
+        vals = {'framework_agreement_id': agreement.id,
+                'unit_cost': price,
+                'total_cost': price * self.proposed_qty,
+                'supplier_id': agreement.supplier_id.id}
+        self.write(vals)
+        res = {}
         if not enough_qty:
```
Comment (yvaucher) [inline framework_agreement_sourcing/model/logistic_requisition_source.py]: This check seems useless as in framework_agreement get_cheapest_agreement_for_qty only returns an agreement if it found a matching one with enough qty.

@nbessi @lepistone  have we somewhere in OCA modules an override of get_cheapest_agreement_for_qty that could return something else than:

(None, None) or (agreement_id, True) ?

Comment (lepistone) [inline framework_agreement_sourcing/model/logistic_requisition_source.py]: @yvaucher I don't know off the top of my head. In any case, ping me if I can help.

**Comment by foutoucour** [comment]
Do you plan to use openerp.fields classes to have a full port to v8.0?
It seems the new api is getting better and better.

also you are using osv and orm. They are a redondant, aren't they?

**Comment by yvaucher** [comment]
@foutoucour I think porting fields in this PR is out of scope of the fix. And odoo 8.0 old api is not yet deprecated. :smiley: 

This PR does not intend to port the whole code to v8.0. I just fixed the onchanges tooking the opportunity to port them to new API.
I agree that v8.0 api is getting better. Nevertheless, after several issues with it, we decided not to port to v8.0 when not necessary.

About `osv` and `orm`:

They are not redundant, as I use direct path for old style fields:
https://github.com/odoo/odoo/blob/8.0/openerp/osv/fields.py

This to use new fields in onchanges. (just seeing I'm calling the wrong datetime BTW)

And didn't changed the old stylish path to Model (which is in fact in models.py now but this is out of scope of this PR)
https://github.com/odoo/odoo/blob/8.0/openerp/osv/orm.py

**Comment by lepistone** [comment]
Thanks! :+1:

**Comment by foutoucour** [comment]
:+1: 
Thanks for the port!

**Pull request merged**